### PR TITLE
Autolathes can now be EMAG'd + Adds Illegal Ammo Design Disk

### PR DESCRIPTION
**Pull request by Skrem7**
# Document the changes in your pull request

You can now EMAG autolathes (even if you don't use harm intent, it'll work and it won't eat the EMAG). This permanently hacks the autolathe (which is indicated by the blue light rapidly flickering in the hack menu).

Furthermore, adds the Illegal Ammo Design Disk to the uplink for 4 TC, which can be slotted into an autolathe to allow it to unlock the ability to print every kind of 10mm and .357 ammo box.

For now, the disk-unique designs are:
For 45000 metal:
- 10mm AP Ammo Box
- 10mm HP Ammo Box
- 10mm Incendiary Ammo Box
- 10mm EMP Ammo Box

For 60000 metal:
- .357 Nutcracker Ammo Box 
- .357 Metalshock Ammo Box
- .357 Heartpiercer Ammo Box
- .357 Wallstake Ammo Box

In addition, the following hacked designs have been added (no disk required: just the blue light on, but the disk will unlock them):
- .357 Ironfeather bullet, 2000 metal (same cost as the individual .357 one)
- .357 Ironfeather Ammo Box, 40000 metal (same cost as the normal .357 ammo box)
- 10mm Soporific Ammo Box, 30000 metal (same cost as the normal 10mm ammo box)

Considering ironfeather/soporific are the same TC cost as the default bullet, and they shouldn't be *too detrimental to give to crew* (one can only go to an already powergaming detective gun, the other would only slot in a surplus carbine if you find one), I don't think this should be too much of an issue.

# Wiki Documentation

EMAG functionality should be documented on the Emag page
Autolathe designs should go on the Designs... page? I forgot what it's called, but I know it exists
New disk should be included on the Syndicate items page

# Changelog

:cl:  
rscadd: Adds the Illegal Ammo Design Disk to the uplink for 4 TC, which allows you to print every type of 10mm and .357 ammo box at the lathe (this includes stuff like 10mm AP, soporific, .357 Heartpiercer, Ironfeather, and more)
rscadd: You can now EMAG autolathes to render them permanently hacked
rscadd: Hacked autolathes can now print individual .357 Ironfeather bullets, as well as 10mm soporific and .357 Ironfeather ammo boxes
/:cl:

### CODE REVIEW

File: code/datums/wires/autolathe.dm [hunk 1]
```diff
@@ -19,15 +19,19 @@
...
 
 /datum/wires/autolathe/on_pulse(wire)
 	var/obj/machinery/autolathe/A = holder
 	switch(wire)
 		if(WIRE_HACK)
-			A.adjust_hacked(!A.hacked)
-			addtimer(CALLBACK(A, /obj/machinery/autolathe.proc/reset, wire), 60)
+			if(!(A.obj_flags & EMAGGED))
+				A.adjust_hacked(!A.hacked)
+				addtimer(CALLBACK(A, /obj/machinery/autolathe.proc/reset, wire), 60)
```
Comment (tattax) [inline code/datums/wires/autolathe.dm]:
```suggestion
				addtimer(CALLBACK(A, TYPE_PROC_REF(/obj/machinery/autolathe, reset), wire), 6 SECONDS)
```

**Comment by wonderinghost** [comment]
Oh no

File: code/datums/wires/autolathe.dm (hunk 1 above)
Comment (Skrem7) [reply code/datums/wires/autolathe.dm]: sure

**Comment by Rdrazga** [comment]
I would prob not have incen or emp be printable since they normally take rare resources to make

**Comment by MajManatee** [comment]
uranium and water, and a special kind of rock that sparks. not really rare or expensive.
anyways this would be dope

**Comment by RG4ORDR** [comment]
This is an extremely dangerous idea that I'd rather not even be a emag tax, but something else like emagging a protolathe after unlocking illegal tech.

**Review by ReddicusDragon — CHANGES_REQUESTED** [review]
Please consider @Rdrazga 's comment on removing incendiary and emp rounds. If you decide they are fine, let me know and I will push this through.

**Review by ReddicusDragon — APPROVED** [review]
I am stupid.

**Pull request merged**